src/node_snapshotable.cc

```diff
@@ -1437,6 +1425,22 @@ void SerializeSnapshotableObjects(Realm* realm,
 
 namespace mksnapshot {
 
+void GetEmbedderEntryFunction(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+  if (!env->embedder_mksnapshot_entry_point()) return;
+  MaybeLocal<Function> jsfn = Function::New(
+      isolate->GetCurrentContext(),
+      [](const FunctionCallbackInfo<Value>& args) {
+        Environment* env = Environment::GetCurrent(args);
+        Local<Value> require_fn = args[0];
```

Comment:
What if `env->embedder_mksnapshot_entry_point` is just the `StartExecutionCallback` we get from the API? Then I think we can just invoke it here using `env->process_object()` and `args[0]` as `StartExecutionCallbackInfo` instead of using the lambdas?

Reply:
I’ve switched to that in a4981baf63a135f361d648b461a61a74d2731d85. Not really sure I have strong feelings either way, but just for context, the original reason for using a lambda here was to decouple the mksnapshot code a bit more from the actual embedder API (it shouldn’t need to worry about what the exact arguments for the callback are).

That being said, as the TODO here indicates, I think it would be nice to refactor this code a bit more anyway so that e.g. we don’t have to store anything on the `Environment` as a way of transferring data to the JS entry point, but that’s something I’d like to tackle as part of a follow-up PR rather than here